app/controllers/song_categories_controller.rb

```diff
@@ -0,0 +1,22 @@
+class SongCategoriesController < ApplicationController
+  load_permissions_and_authorize_resource
+
+  def index
+    @song_categories = initialize_grid(SongCategory.all, order: 'name', order_direction: 'asc')
+  end
+
+  def show #???
+    Song.increment_counter(:visits, @song)
+    @popular = Song.by_visits
+  end
+
+  def search
```

Comment:
You might not need the search and search params since these are used with the sidebar in the song view. It allows searching for songs directly and is used in conjunction with JS. Since this is probably not desirable with the categories you can probably remove these methods.

Reply:
Removed